docs/trestle_author.md

```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
-```
+  However, violations such as adding or removing a heading at the top level is not acceptable:
 
-The following heading would be acceptable.
+  ```markdown
+  # Template heading 1
+  Content for heading one
+  ## Non-required sub header
+  Content for non-required sub header
+  # Template heading 2
+  Content for heading two
+  ## Template sub heading
+  Content for template sub heading
+  # Top level heading that is not okay <<< NOT OKAY
+  ```
 
-```markdown
-# heading for strict enforcement
-my_key: my value
-my_key_2: my value
-my_other key with strange stuff?? my value
-```
+  For each of the headings - the text of the heading is enforced with one caveat:
 
-This capability, where available, is activated by `--governed-heading` or `-gh`
+  - If the template heading text is wrapped in curly brackets `{}` then the name is not measured e.g. `# {Insert title here}`.
 
-### Drawio enforcement mechanisms
+  ## Drawio enforcement mechanisms
 
-Drawio or [diagrams.net](https://app.diagrams.net/) is a diagramming platform which has significant use for architecture diagrams. In the context of governance of content, trestle is supporting enforcement of metadata.
+  Drawio or [diagrams.net](https://app.diagrams.net/) is a diagramming platform which has significant use for architecture diagrams. In the context of governance of content, trestle is supporting enforcement of metadata.
 
-Drawio (or `mxgraph`) files have a set of data fields. In a drawio file this is available in the edit menu as *edit data*. The diagram below shows how to access the (meta)data.
+  Drawio (or `mxgraph`) files have a set of data fields. In a drawio file this is available in the edit menu as *edit data*. The diagram below shows how to access the (meta)data.
 
-![Accessing the drawio data editor](assets/drawio_data_menu.png "Accessing the drawio data editor")
+  ![Accessing the drawio data editor](assets/drawio_data_menu.png "Accessing the drawio data editor")
 
-The data presents as a set of key-value pairs which can be edited (see below). The data is bound to each tab in a drawio file. The trestle CLI currently expects that metadata (whether from the template or file to be measured) is in the first tab when editing the draw io file.
+  The data presents as a set of key-value pairs which can be edited (see below). The data is bound to each tab in a drawio file. The trestle CLI currently expects that metadata (whether from the template or file to be measured) is in the first tab when editing the draw io file.
 
-![Editing drawio data](assets/drawio_editing_data.png "Editing drawio data")
+  ![Editing drawio data](assets/drawio_editing_data.png "Editing drawio data")
 
-## Template versioning
+  </details>
 
-### Prerequisite:
+- <details>
+  <summary><b>Setting Up Templates and Documents Via CLI</b></summary>
+  Trestle allows to set up templates and governed documents in a several different ways based on the nature of the needed governance.
```

Comment:
Trestle allows the setup of templates and ...

Reply:
Done